baxter_collaboration_lib/test/test_ros_thread_image.cpp

```diff
@@ -16,24 +16,70 @@ class ROSThreadImageTester
         image_pub = it.advertise("/"+name+"/image", 1);
     }
 
-    void sendTestImage()
+    void sendTestImage(std::string encoding = "bgr8")
     {
-        // Let's create a 200x200 black image with a red circle in the center
-        cv::Mat img(200, 200, CV_8UC3, cv::Scalar(0,0,0));
-        cv::circle(img, cv::Point(100, 100), 20, CV_RGB(255,0,0), -1);
+        // Let's create a 200x200 black image with a red circle
+        // (white cirlce for mono / 1-channel images) in the center
+        // Checks encoding and correctly creates an image with
+        // the requested encoding and publishes it
+        if (encoding == "mono8")
+        {
+            cv::Mat img(200, 200, CV_8UC1, cv::Scalar(0,0,0));
+            cv::circle(img, cv::Point(100, 100), 20, CV_RGB(255,255,255), -1);
+            sensor_msgs::ImagePtr msg = cv_bridge::CvImage(std_msgs::Header(), encoding, img).toImageMsg();
+            image_pub.publish(msg);       
+        }
+        /*
+        if (encoding == "mono16")
+        {
+            cv::Mat img(200, 200, CV_16UC1, cv::Scalar(0,0,0));
+            cv::circle(img, cv::Point(100, 100), 20, CV_RGB(255,0,0), -1);
+            sensor_msgs::ImagePtr msg = cv_bridge::CvImage(std_msgs::Header(), encoding, img).toImageMsg();
+            image_pub.publish(msg);        
+        }
+
+        if (encoding == "rgb8")
+        {
+            cv::Mat img(200, 200, CV_8UC3, cv::Scalar(0,0,0));
+            cv::circle(img, cv::Point(100, 100), 20, CV_RGB(255,0,0), -1);
+            sensor_msgs::ImagePtr msg = cv_bridge::CvImage(std_msgs::Header(), encoding, img).toImageMsg();
+            image_pub.publish(msg);       
+        }
+
+        if (encoding == "bgra8")
+        {
+            cv::Mat img(200, 200, CV_8UC4, cv::Scalar(0,0,0));
+            cv::circle(img, cv::Point(100, 100), 20, CV_RGB(255,0,0), -1);
+            sensor_msgs::ImagePtr msg = cv_bridge::CvImage(std_msgs::Header(), encoding, img).toImageMsg();
+            image_pub.publish(msg);       
+        }
+
+        if (encoding == "rgba8")
+        {
+            cv::Mat img(200, 200, CV_8UC4, cv::Scalar(0,0,0));
+            cv::circle(img, cv::Point(100, 100), 20, CV_RGB(255,0,0), -1);
+            sensor_msgs::ImagePtr msg = cv_bridge::CvImage(std_msgs::Header(), encoding, img).toImageMsg();
+            image_pub.publish(msg);        
+        }  */
+        else
```

Comment:
add an explicit `else if (encoding == "bgr8")` so that it is clear to the user, plus an `else` ERROR